action/protocol/rewarding/protocol.go

```diff
@@ -396,28 +419,35 @@ func (p *Protocol) settleAction(
 			return nil, err
 		}
 	}
-	gasFee := big.NewInt(0).Mul(actionCtx.GasPrice, big.NewInt(0).SetUint64(actionCtx.IntrinsicGas))
-	depositLog, err := DepositGas(ctx, sm, gasFee)
-	if err != nil {
-		return nil, err
-	}
-	if depositLog != nil {
-		tLogs = append(tLogs, depositLog)
-	}
-	if err := p.increaseNonce(sm, actionCtx.Caller, actionCtx.Nonce); err != nil {
-		return nil, err
+	if !isSystemAction || !protocol.MustGetFeatureCtx(ctx).SkipUpdateForSystemAction {
```

Comment:
can we move `ctx.SkipUpdateForSystemAction` flag and related changes here to another PR?